src/components/Wallet/WalletGenerator.jsx

```diff
@@ -143,18 +143,20 @@ class WalletGenerator extends React.Component {
     ).length;
   };
 
-  toggleImporters = (event) => {
+  toggleImporters = (event, type) => {
     event.preventDefault();
     const {
       setImportersVisible,
       configuring,
       resetWallet,
+      initialLoadComplete,
       setGenerating,
     } = this.props;
 
     if (!configuring) {
       setGenerating(false);
-      resetWallet();
+      if (type === "edit") initialLoadComplete();
+      if (type === "clear") resetWallet();
```

Comment:
Perhaps use a boolean instead?  

```js
toggleImporters = (event, clear=true)
```

Reply:
Thanks for the review, @waldenraines! Switched it to a boolean as suggested.